{
  "abstract": [
    {
      "text": "Instructs the test to expect a failure in an upcoming assertion.",
      "type": "text"
    }
  ],
  "documentVersion": 0,
  "hierarchy": {
    "paths": [
      [
        "doc://com.apple.documentation/documentation/technologies",
        "doc://com.apple.documentation/documentation/xctest",
        "doc://com.apple.documentation/documentation/xctest/expected-failures"
      ]
    ]
  },
  "identifier": {
    "interfaceLanguage": "occ",
    "url": "doc://com.apple.documentation/documentation/xctest/3726077-xctexpectfailure"
  },
  "kind": "symbol",
  "legacy_identifier": 3726077,
  "legalNotices": {
    "copyright": "Copyright &copy; 2025 Apple Inc. All rights reserved.",
    "privacyPolicy": "https://www.apple.com/privacy/privacy-policy",
    "termsOfUse": "https://www.apple.com/legal/internet-services/terms/site.html"
  },
  "metadata": {
    "externalID": "c:@F@XCTExpectFailure",
    "modules": [
      {
        "name": "XCTest"
      }
    ],
    "platforms": [
      {
        "current": "16.3",
        "introducedAt": "12.5",
        "name": "Xcode"
      }
    ],
    "role": "symbol",
    "roleHeading": "Function",
    "symbolKind": "func",
    "title": "XCTExpectFailure"
  },
  "primaryContentSections": [
    {
      "declarations": [
        {
          "languages": [
            "occ"
          ],
          "platforms": [
            "iOS",
            "iPadOS",
            "macOS",
            "tvOS",
            "visionOS",
            "watchOS"
          ],
          "tokens": [
            {
              "kind": "keyword",
              "text": "void"
            },
            {
              "kind": "text",
              "text": " "
            },
            {
              "kind": "identifier",
              "text": "XCTExpectFailure"
            },
            {
              "kind": "text",
              "text": "("
            },
            {
              "identifier": "doc://com.apple.documentation/documentation/foundation/nsstring",
              "kind": "typeIdentifier",
              "preciseIdentifier": "c:objc(cs)NSString",
              "text": "NSString"
            },
            {
              "kind": "text",
              "text": " *failureReason);"
            }
          ]
        }
      ],
      "kind": "declarations"
    }
  ],
  "references": {
    "doc://com.apple.documentation/documentation/foundation/nsstring": {
      "identifier": "doc://com.apple.documentation/documentation/foundation/nsstring",
      "kind": "symbol",
      "role": "symbol",
      "title": "NSString",
      "type": "topic",
      "url": "/documentation/foundation/nsstring"
    },
    "doc://com.apple.documentation/documentation/technologies": {
      "identifier": "doc://com.apple.documentation/documentation/technologies",
      "kind": "technologies",
      "title": "Technologies",
      "type": "topic",
      "url": "/documentation/technologies"
    },
    "doc://com.apple.documentation/documentation/xctest": {
      "identifier": "doc://com.apple.documentation/documentation/xctest",
      "kind": "symbol",
      "role": "collection",
      "title": "XCTest",
      "type": "topic",
      "url": "/documentation/xctest"
    },
    "doc://com.apple.documentation/documentation/xctest/3726077-xctexpectfailure": {
      "abstract": [
        {
          "text": "Instructs the test to expect a failure in an upcoming assertion.",
          "type": "text"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/xctest/3726077-xctexpectfailure",
      "kind": "symbol",
      "role": "symbol",
      "title": "XCTExpectFailure",
      "type": "topic",
      "url": "/documentation/xctest/3726077-xctexpectfailure"
    },
    "doc://com.apple.documentation/documentation/xctest/3726078-xctexpectfailureinblock": {
      "abstract": [
        {
          "text": "Instructs the test to expect a failure in an assertion in the provided block of code.",
          "type": "text"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/xctest/3726078-xctexpectfailureinblock",
      "kind": "symbol",
      "role": "symbol",
      "title": "XCTExpectFailureInBlock",
      "type": "topic",
      "url": "/documentation/xctest/3726078-xctexpectfailureinblock"
    },
    "doc://com.apple.documentation/documentation/xctest/3726079-xctexpectfailurewithoptions": {
      "abstract": [
        {
          "text": "Instructs the test to expect a failure in an upcoming assertion, with options to customize expected failure checking and handling.",
          "type": "text"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/xctest/3726079-xctexpectfailurewithoptions",
      "kind": "symbol",
      "role": "symbol",
      "title": "XCTExpectFailureWithOptions",
      "type": "topic",
      "url": "/documentation/xctest/3726079-xctexpectfailurewithoptions"
    },
    "doc://com.apple.documentation/documentation/xctest/3726080-xctexpectfailurewithoptionsinblo": {
      "abstract": [
        {
          "text": "Instructs the test to expect a failure in an assertion in the provided block of code, with options to customize expected failure checking and handling.",
          "type": "text"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/xctest/3726080-xctexpectfailurewithoptionsinblo",
      "kind": "symbol",
      "role": "symbol",
      "title": "XCTExpectFailureWithOptionsInBlock",
      "type": "topic",
      "url": "/documentation/xctest/3726080-xctexpectfailurewithoptionsinblo"
    },
    "doc://com.apple.documentation/documentation/xctest/expected-failures": {
      "identifier": "doc://com.apple.documentation/documentation/xctest/expected-failures",
      "kind": "article",
      "role": "collectionGroup",
      "title": "Expected Failures",
      "type": "topic",
      "url": "/documentation/xctest/expected-failures"
    },
    "doc://com.apple.documentation/documentation/xctest/xctexpectedfailure": {
      "abstract": [
        {
          "text": "An object that represents an expected test failure.",
          "type": "text"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/xctest/xctexpectedfailure",
      "kind": "symbol",
      "role": "symbol",
      "title": "XCTExpectedFailure",
      "type": "topic",
      "url": "/documentation/xctest/xctexpectedfailure"
    },
    "doc://com.apple.documentation/documentation/xctest/xctexpectedfailure/options": {
      "abstract": [
        {
          "text": "Options that determine how the test matches the expected failure to an actual test failure, and whether an unfulfilled expected failure results in a test failure.",
          "type": "text"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/xctest/xctexpectedfailure/options",
      "kind": "symbol",
      "role": "symbol",
      "title": "XCTExpectedFailure.Options",
      "type": "topic",
      "url": "/documentation/xctest/xctexpectedfailure/options"
    }
  },
  "schemaVersion": {
    "major": 0,
    "minor": 3,
    "patch": 0
  },
  "sections": [],
  "seeAlsoSections": [
    {
      "generated": true,
      "identifiers": [
        "doc://com.apple.documentation/documentation/xctest/xctexpectedfailure",
        "doc://com.apple.documentation/documentation/xctest/xctexpectedfailure/options",
        "doc://com.apple.documentation/documentation/xctest/3726078-xctexpectfailureinblock",
        "doc://com.apple.documentation/documentation/xctest/3726079-xctexpectfailurewithoptions",
        "doc://com.apple.documentation/documentation/xctest/3726080-xctexpectfailurewithoptionsinblo"
      ],
      "title": "Expected Failures"
    }
  ],
  "variants": [
    {
      "paths": [
        "documentation/xctest/3726077-xctexpectfailure"
      ],
      "traits": [
        {
          "interfaceLanguage": "occ"
        }
      ]
    }
  ]
}
